parser/v2/elementparser.go

```diff
@@ -411,6 +411,23 @@ var element elementParser
 
 type elementParser struct{}
 
+const maxIgnoreNewLineTagLength = 8
+
+var ignoreNewLinesTags = []string{
+	"b",
+	"strong",
+	"i",
+	"em",
+	"span",
+	"u",
+	"br",
+	"abbr",
+	"code",
+	"mark",
+	"small",
+	"q",
+}
```

Comment:
My first thought is that this seems to be a list of inline elements and that the intent is that inline elements shouldn't be placed on a new line, unless the previous node is a text node, and ends with whitespce (since in this case, we're free to add more space, because it will be collapsed to a single space due to whitespace normalisation rules).

https://www.w3schools.com/html/html_blocks.asp

My second thought is that maybe it would be faster to use a dictionary lookup instead of iterating through the loop of elements.

Something like this, but for inline:

```
var blockElements = map[string]struct{}{
	"address": {}, "article": {}, "aside": {}, "body": {}, "blockquote": {}, "canvas": {}, "dd": {}, "div": {}, "dl": {}, "dt": {}, "fieldset": {}, "figcaption": {}, "figure": {}, "footer": {}, "form": {}, "h1": {}, "h2": {}, "h3": {}, "h4": {}, "h5": {}, "h6": {}, "head": {}, "header": {}, "hr": {}, "html": {}, "li": {}, "main": {}, "meta": {}, "nav": {}, "noscript": {}, "ol": {}, "p": {}, "pre": {}, "script": {}, "section": {}, "table": {}, "template": {}, "tfoot": {}, "turbo-stream": {}, "ul": {}, "video": {},
}

func (e Element) isBlockElement() bool {
	_, ok := blockElements[e.Name]
	return ok
}
```

Reply:
Oh thank you I didn't know that on w3schools there was a page clearly given Inline and Block Elements. I will so change the list to been accord with inline element cause they are the only that must subject to this change.

For the dictionary thing. I didn't think first cause for me Dictionary is for key value pair but that clever !
I will use that rather then an array ^^